libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routefetch/RoutesViewModel.kt

```diff
@@ -11,24 +12,38 @@ import com.mapbox.navigation.base.route.NavigationRouterCallback
 import com.mapbox.navigation.base.route.RouterFailure
 import com.mapbox.navigation.base.route.RouterOrigin
 import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
 import com.mapbox.navigation.dropin.lifecycle.UIViewModel
-
-sealed class RoutesAction {
-    data class FetchPoints(val points: List<Point>) : RoutesAction()
-    data class FetchOptions(val options: RouteOptions) : RoutesAction()
-    data class SetRoutes(val routes: List<NavigationRoute>, val legIndex: Int = 0) : RoutesAction()
-}
+import com.mapbox.navigation.dropin.model.Destination
+import com.mapbox.navigation.utils.internal.toPoint
+import kotlinx.coroutines.flow.StateFlow
 
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
-internal class RoutesViewModel : UIViewModel<Unit, RoutesAction>(Unit) {
+internal class RoutesViewModel(
+    private val navigationState: StateFlow<NavigationState>,
+    private val locationState: StateFlow<Location?>
```

Comment:
We have a goal to be able to expose our view model implementations publically.

Ideally we delete these 2 lines. It's better to depend on the view models (e.g., NavigationStateViewModel and LocationViewModel)

But we also, want to minimize the dependencies. So best case, we are able to remove all the dependencies from this view model.